doc/plugin_server_nodeattestor_gcp_instance_identity_token.md

```diff
@@ -0,0 +1,14 @@
+# Server plugin: NodeAttestor "gcp_instance_identity_token"
+
+*Must be used in conjunction with the agent-side gcp_instance_identity_token plugin*
+
+The `gcp_instance_identity_token` plugin automatically attests instances using the [GCP Instance Identity Token](https://cloud.google.com/compute/docs/instances/verifying-instance-identity). It also allows an operator to use GCP Instance IDs when defining SPIFFE ID attestation policies. 
+Agents attested by the gcp_instance_identity_token attestor will be issued a SPIFFE ID like `spiffe://example.org/agent/gcp_instance_identity_token/PROJECT_ID/INSTANCE_ID`
+
+| Configuration           | Description                                                                                        | Default                                    |
+|-------------------------|----------------------------------------------------------------------------------------------------|--------------------------------------------|
+| trust_domain            | The trust domain that the agent belongs to.                                                        |                                            |
+| audience                | Audience parameter used inside the tokens. Needs be equal to the audience parameter in the agent.  |                                            |
+| google_cert_url         | URL of the Google certificates used to sign the Identity Token                                     | https://www.googleapis.com/oauth2/v1/certs |
+| max_token_lifetime_secs | Maximum token lifetime in seconds for the token to be considered valid. Default: 1 day             | 86400                                      |
+| clock_skew_secs         | Leeway during token validity timeframe validation in seconds. Default: 5 minutes                   | 300                                        |
```

Comment:
This project makes a (perhaps questionable) assumption that clocks are reasonably synchronized... else we run into all sorts of problems with x.509 validation. I would suggest to 1) tighten this to 60s or something smaller, and 2) maybe hardcode it... the fewer knobs the better. I guess depending on the answer to my question about `max_token_lifetime_secs`, this might change

Reply:
I'm ok with this assumption. Are these assumptions documented somewhere?

I'll set it the parameter to 60s and remove the configuration option.

Reply:
> Are these assumptions documented somewhere?

They are not, but probably should be. That one popped up during a community call in which we were trying to think if we could solve clock-based attacks. As a user, where would you expect to find this documented?

Reply:
I would like to see a design decision document in the Spire documentation that describes assumptions and decisions that were made  to get an insight into the considerations that led to the current state of the product.